crates/api/src/site/mod_log.rs

```diff
@@ -42,69 +53,268 @@ impl Perform for GetModlog {
+        blocking(context.pool(), move |conn| {
+          ModRemoveCommentView::list(
+            conn,
+            community_id,
+            mod_person_id,
+            other_person_id,
+            page,
+            limit,
+            hide_modlog_names,
+          )
+        })
+        .await??
+      }
+      _ => Default::default(),
+    };
 
-    let banned_from_community = blocking(context.pool(), move |conn| {
-      ModBanFromCommunityView::list(conn, community_id, mod_person_id, page, limit)
-    })
-    .await??;
+    let banned_from_community = match type_ {
+      ModlogActionType::All | ModlogActionType::ModBanFromCommunity => {
+        blocking(context.pool(), move |conn| {
+          ModBanFromCommunityView::list(
+            conn,
+            community_id,
+            mod_person_id,
+            other_person_id,
+            page,
+            limit,
+            hide_modlog_names,
+          )
+        })
+        .await??
+      }
+      _ => Default::default(),
+    };
 
-    let added_to_community = blocking(context.pool(), move |conn| {
-      ModAddCommunityView::list(conn, community_id, mod_person_id, page, limit)
-    })
-    .await??;
+    let added_to_community = match type_ {
+      ModlogActionType::All | ModlogActionType::ModAddCommunity => {
+        blocking(context.pool(), move |conn| {
+          ModAddCommunityView::list(
+            conn,
+            community_id,
+            mod_person_id,
+            other_person_id,
+            page,
+            limit,
+            hide_modlog_names,
+          )
+        })
+        .await??
+      }
+      _ => Default::default(),
+    };
 
-    let transferred_to_community = blocking(context.pool(), move |conn| {
-      ModTransferCommunityView::list(conn, community_id, mod_person_id, page, limit)
-    })
-    .await??;
+    let transferred_to_community = match type_ {
+      ModlogActionType::All | ModlogActionType::ModTransferCommunity => {
+        blocking(context.pool(), move |conn| {
+          ModTransferCommunityView::list(
+            conn,
+            community_id,
+            mod_person_id,
+            other_person_id,
+            page,
+            limit,
+            hide_modlog_names,
+          )
+        })
+        .await??
+      }
+      _ => Default::default(),
+    };
 
-    let hidden_communities = blocking(context.pool(), move |conn| {
-      ModHideCommunityView::list(conn, community_id, mod_person_id, page, limit)
-    })
-    .await??;
+    let hidden_communities = match type_ {
+      ModlogActionType::All | ModlogActionType::ModHideCommunity => {
+        if other_person_id.is_none() {
+          blocking(context.pool(), move |conn| {
+            ModHideCommunityView::list(
+              conn,
+              community_id,
+              mod_person_id,
+              page,
+              limit,
+              hide_modlog_names,
+            )
+          })
```

Comment:
It looks like all of these methods have exactly the same parameters. Please make a struct for them, and then you just need to pass a single param for each of these method calls. Should shorten the code significantly.